### PR TITLE
refactor: Fix implicit value conversion in  formatPingTime

### PR DESCRIPTION
**Pull request by Empact**
Alternative to #18252, see motivation there.

This changes `CNodeStats` to handle ping timestamps as their original incoming usec `int64_t` values until the time they need to be displayed.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -165,12 +165,12 @@ static UniValue getpeerinfo(const JSONRPCRequest& request)
...
         obj.pushKV("conntime", stats.nTimeConnected);
         obj.pushKV("timeoffset", stats.nTimeOffset);
-        if (stats.dPingTime > 0.0)
-            obj.pushKV("pingtime", stats.dPingTime);
-        if (stats.dMinPing < static_cast<double>(std::numeric_limits<int64_t>::max())/1e6)
-            obj.pushKV("minping", stats.dMinPing);
-        if (stats.dPingWait > 0.0)
-            obj.pushKV("pingwait", stats.dPingWait);
+        if (stats.m_ping_usec > 0)
+            obj.pushKV("pingtime", stats.m_ping_usec / 1e6);
```
Comment (promag) [inline src/rpc/net.cpp]: Should cast to double first?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

**Comment by fanquake** [comment]
cc @vasild

File: src/rpc/net.cpp (hunk 1 above)
Comment (Empact) [reply src/rpc/net.cpp]: Done

**Review by promag — COMMENTED** [review]
ACK 1891245e7318bf625bbf67aab08a79fc3e87b61d, added cast to double and also braces.

File: src/rpc/net.cpp (hunk 1 above)
Comment (vasild) [reply src/rpc/net.cpp]: Assuming `x` is `int64_t`, then all of these are equivalent:

```
x / 1e6;
(double)x / 1e6;
((double)x) / 1e6;
x / 1000000.0;
```

**Comment by vasild** [comment]
ACK 1891245

I tested this with Clang 10 and [the warning](https://github.com/bitcoin/bitcoin/pull/18252#issue-382935046) is gone.

Nit: the cast to `double` added in 1891245 is not necessary (I am ok with it).

File: src/rpc/net.cpp (hunk 1 above)
Comment (promag) [reply src/rpc/net.cpp]: Indeed, I tend to prefer the explicit numerator cast though.

**Comment by practicalswift** [comment]
ACK 1891245e7318bf625bbf67aab08a79fc3e87b61d -- patch looks correct

**Comment by MarcoFalke** [comment]
Would be nice to add (functional) tests for this, since it is uncovered right now.

**Pull request merged**